[DOCS] Update pass infra tutorial

zhiics opened the pull request:
#6144 

This PR updates the tutorial for the pass manager. It fixes some of the stale terminology used in tutorial to reflect the support for both Relay and tir. In addition, the examples for customizing optimization pipeline and debugging are moved to a separate tutorial and included in the how-to doc.

zhiics commented:
cc @tqchen @jroesch

tqchen commented on docs/dev/use_pass_infra.rst:
```diff
@@ -0,0 +1,117 @@
+
+
+For more pass infra related examples in Python and C++, please refer to
+`tests/python/relay/test_pass_manager.py`_ and
```
Shall we put some examples in the tvm.ir.Sequential etc and refer to the API docs?

It seems to be not too useful to refer to the test code

zhiics replied:
I realized that we have a tutorial already https://tvm.apache.org/docs/tutorials/dev/relay_pass_infra.html

We can probably just refactor this one and add it to how-to

tqchen commented on docs/dev/use_pass_infra.rst:
```diff
@@ -0,0 +1,117 @@
+            print(mod)
+
+    with relay.build_config(opt_level=3, trace=print_ir):
+            with tvm.target.create("llvm"):
```
alignment

tqchen commented on docs/dev/use_pass_infra.rst:
```diff
@@ -0,0 +1,117 @@
+            print(f"Running pass: {}", info)
+            print(mod)
+
+    with relay.build_config(opt_level=3, trace=print_ir):
```
use tvm.transform.PassContext

tqchen commented on docs/dev/use_pass_infra.rst:
```diff
@@ -0,0 +1,117 @@
+            mod = seq(mod)
+
+Debugging
+=========
```
yse second level title as opposed to all the first level title.

tqchen commented on docs/dev/use_pass_infra.rst:
```diff
@@ -0,0 +1,117 @@
+The following example not only illustrates how users can directly create a sequential
+pass using Python APIs (this could be applied to module- and function-level
+passes as well), but also explains how we can build an optimization pipeline
+using ``Sequential`` associated with other types of passes.
```
py:class:`tvm.transform.Sequantial`

tqchen commented on docs/dev/use_pass_infra.rst:
```diff
@@ -0,0 +1,117 @@
+can use some of the more advanced features in the pass infra, e.g. customizing your
+own optimization pass pipeline and debugging. Here, we will mainly focus on
+using sequential passes. For more details about other type of passes and how
+they interact with ``Sequential``, please refer to :ref:`pass-infra`.
```
Consider add a summary section in the end to summarize the key points

zhiics commented:
@tqchen I moved the debugging section to the the tutorial and updated the tutorial correspondingly. Since there are examples using different ways (including sequential) to invoke passes, I removed it from how-to and only kept the tutorial. I think that should be sufficient.
